src/globus_sdk/tokenstorage/sqlite_adapter.py

```diff
@@ -0,0 +1,217 @@
+import json
+import sqlite3
+from typing import Any, Dict, Mapping, Optional, cast
+
+from globus_sdk.auth import OAuthTokenResponse
+from globus_sdk.tokenstorage.base import FileAdapter
+from globus_sdk.version import __version__
+
+
+class SQLiteAdapter(FileAdapter):
+    """
+    :param dbname: the name of the DB file to write to and read from
```

Comment:
Would be good to mention that passing dbname as :memory: has special behavior here.

I almost would rather we didn't rely on magic string behavior, and instead had some in_memory boolean param to pass but we're inheriting the :memory: usage from sqlite so its probably fine.

Reply:
I'll do a follow-up PR to add that to the doc here.

Given that `:memory:` is specially treated by sqlite itself -- not merely the python sqlite bindings -- I think it's least confusing to stick with this behavior, at least for anyone familiar with sqlite. If I were designing an interface for something backed by a file or memory in a vacuum, without the sqlite context, I'd also probably choose something different.